server/src/main/resources/com/thoughtworks/go/server/dao/maps/JobInstance.xml

```diff
@@ -83,55 +71,46 @@
         <parameter property="rerun"/>
     </parameterMap>
 
-    <cacheModel id="latest-completed-job-instances" type="MEMORY" readOnly="true">
-        <flushInterval seconds="300"/>
-        <flushOnExecute statement="insertBuild"/>
-        <flushOnExecute statement="ignoreBuildById"/>
-        <flushOnExecute statement="updateResult"/>
-        <flushOnExecute statement="updateStatus"/>
-        <property name="cache-size" value="100" />
-    </cacheModel>
-
-    <select id="latestCompletedJobs" resultMap="select-build-with-identifier" cacheModel="latest-completed-job-instances">
+    <select id="latestCompletedJobs" resultMap="select-build-with-identifier" useCache="true">
```

Comment:
How does this cache get invalidated? The older cacheModel used to get invalidated on inserting of new builds or when jobs completed. I dont see us doing a `flushCache` anywhere.

Reply:
Also, as per docs, `useCache` is set to `true` by default for all select statements, so don't we need to take care of invalidating them too? Or should we just turn to `false`?

Reply:
ibatis would allow you to configure cache "namespaces" by declaring `<cacheModel/>` elements and tying them to specific SQLs declared in the mapper XML.

mybatis prefers a single cache for the entire mapper file. This can be enabled by adding a `<cache/>` element at the top of the file (something we've not done). We selectively turn on caching for specific statements (since that was the original ibatis behavior as well). The cache is busted on any `insert`/`update`/`delete` operation that happens — so in a way we're covered. The downside of this approach is that the cache is busted a bit more aggressively. A work around is to put SQL statements that affect the same cache in the same XML file, but that causes other sorts of issues when you'd want to bust 2 caches when a particular record is updated/added/deleted, for e.g. This is a problem we don't face, currently.

Reply:
Ignore what I said above, I *think* i may be mistaken.

Reply:
As I understand from the docs, without the `<cache/>` element, L2 cache is not enabled, so its just the session level cache coming into play, which is not the equivalent of what we had earlier. correct? For instance, `latestCompletedJobs` (in JobInstance.xml) gets called too often, and not caching it would be expensive (in theory). 

Given, we used cacheModels at very few places, I was also thinking along the lines of getting rid of it. Meaning - what if we did not rely on the mybatis L2 cache at all? We use ehcache (`GoCache`) in most places, why not use the same for the statements that used the cachemodels earlier and turn off L2 cache for mybatis? Handling cache invalidation is pretty easy in code especially since we know the exact statements which used to cause these cache entries to get invalidated earlier. We wouldn't have to mess around with the mapping files at all in that case.  WDYT?